nats/src/main.rs

```diff
@@ -23,18 +23,32 @@ const ENV_NATS_CLIENT_SEED: &str = "CLIENT_SEED";
 fn main() -> Result<(), Box<dyn std::error::Error>> {
     // handle lattice control messages and forward rpc to the provider dispatch
     // returns when provider receives a shutdown control message
-    provider_main(
-        NatsMessagingProvider::default(),
-        Some("Nats Messaging Provider".to_string()),
-    )?;
+    let host_data = load_host_data()?;
+    let provider = if let Some(c) = host_data.config_json.as_ref() {
+        // fall back to the safe default if the JSON is bad
+        let config: ConnectionConfig = match serde_json::from_str(c) {
+            Ok(c) => c,
+            Err(e) => {
+                error!("Failed to parse host data config JSON: {e:?}");
+                ConnectionConfig::default()
```

Comment:
Should this be a fatal error? I'm not sure a user would want the provider to start with default config if the config they try to pass is malformed

Reply:
agreed